sdk/nodejs/dagger/connect.ts

```diff
@@ -0,0 +1,147 @@
+// WIP(TomChv): This file shall be renamed to something else
+import { Client } from './client';
+import { execa, execaCommandSync } from 'execa';
+import path from 'path';
+import axios from 'axios';
+
+/**
+ * Cloak binary name
+ */
+const CLOAK_BINARY = "cloak";
+
+/**
+ * ConnectOpts defines option used to run cloak
+ * in dev mode.
+ * Options are based on `dagger cloak` CLI.
+ */
+export interface ConnectOpts {
+	LocalDirs?: Record<string, string>;
+	Port?: number;
+	Workdir?: string;
+	ConfigPath?: string;
+}
+
+/**
+ * connect runs cloak GraphQL server and initializes a
+ * Dagger client.
+ * This implementation is based on the existing Go SDK.
+ */
+export async function connect(config: ConnectOpts): Promise<Client> {
+	// exit with error if we are not using the non-Cloak dagger binary (< 0.3.0)
+	await verifyCloakBinary();
+
+	// Create config with default values that may be overridden
+	// by config if values are set.
+	const _config: Required<ConnectOpts> = {
```

Comment:
nit: Any reason of using the underscore in front of config ? Why not using another name like defaultConfig ? AFAIK the underscore is a convention to declare private variables.